### PR TITLE
Remove redundant field from GetDecommissionStateResponse

### LINKED ISSUE #4750: [BUG] [Zone Decommission] GET response has redundant field
**Describe the bug**
The current GET API call to get the decommission state is - 
`curl -X GET "localhost:9202/_cluster/decommission/awareness/_status?pretty"`

```
{
	"awareness" : {
		"zone" : "zone-1"
	},
	"status" : "SUCCESSFUL"
}
```

Here, since `awareness` is already mentioned in GET path, the `awareness` field in GET response looks redundant. 

**To Reproduce**
1. Decommission an attribute - `curl -X PUT "localhost:9201/_cluster/decommission/awareness/zone/zone-2?pretty"`
2. Get Decommission status - `curl -X GET "localhost:9201/_cluster/decommission/awareness/_status?pretty"`

**Expected behavior**
We can simplify the response to just have attribute name and status as field. Something like -

```
> curl -X GET "localhost:9202/_cluster/decommission/awareness/_status?pretty"
{
  "zone" : "zone-1",
  "status" : "SUCCESSFUL"
}
```

### PR DESCRIPTION
**Pull request by imRishN**
Signed-off-by: Rishab Nahata <rnnahata@amazon.com>

### Description
This PR tries to remove the redundant field from GET api to retrieve decommission state

Before the fix - 
```
curl -X GET "localhost:9202/_cluster/decommission/awareness/_status?pretty"

{
	"awareness" : {
		"zone" : "zone-1"
	},
	"status" : "SUCCESSFUL"
}
```

After the fix - 

```
~                                                                                                                                                                                                                  03:23:31
> curl -X GET "localhost:9200/_cluster/decommission/awareness/zone/_status?pretty"
{ }

~                                                                                                                                                                                                                  03:23:36
> curl -X PUT "localhost:9201/_cluster/decommission/awareness/zone/zone-2?pretty"
{
  "acknowledged" : true
}

~                                                                                                                                                                                                                  03:23:58
> curl -X GET "localhost:9200/_cluster/decommission/awareness/zone/_status?pretty"
{
  "zone-2" : "SUCCESSFUL"
}

~                                                                                                                                                                                                                  03:24:03
> curl -X GET "localhost:9200/_cluster/decommission/awareness/random/_status?pretty"
{ }

~                                                                                                                                                                                                                  03:24:11
> curl -X GET "localhost:9202/_cluster/decommission/awareness/random/_status?pretty"
{ }

~                                                                                                                                                                                                                  03:24:18
> curl -X GET "localhost:9202/_cluster/decommission/awareness/zone/_status?pretty"
{
  "zone-2" : "SUCCESSFUL"
}

~                                                                                                                                                                                                                  03:24:24
> curl -X GET "localhost:9202/_cluster/decommission/awareness/_status?pretty"
{
  "error" : "no handler found for uri [/_cluster/decommission/awareness/_status] and method [GET]"
}
```

### Issues Resolved
#4750 

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by gbbafna — APPROVED** [review]

**Comment by sachinpkale** [comment]
Overall changes look good. Can you please update the PR description with API response before and after the change?

Also, the build is failing. Please fix.

**Review by sachinpkale — APPROVED** [review]

**Review by saikaranam-amazon — APPROVED** [review]
LGTM - Can we change the RFC to reflect the latest contract?

**Review by Bukhtawar — COMMENTED** [review]
Lets add support for per attribute GET

**Comment by imRishN** [comment]
@Bukhtawar as discussed, updated the Request path and Response for the GET request. Now, the path will need to have `attributeName` specified. The response for a valid request will be `attributeValue: status`

File: server/src/main/java/org/opensearch/action/admin/cluster/decommission/awareness/get/GetDecommissionStateRequest.java [hunk 1]
```diff
@@ -10,31 +10,71 @@
...
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
+        out.writeString(attributeName);
     }
 
     @Override
     public ActionRequestValidationException validate() {
-        return null;
+        ActionRequestValidationException validationException = null;
+        if (attributeName == null || Strings.isEmpty(attributeName)) {
```
Comment (Bukhtawar) [inline server/src/main/java/org/opensearch/action/admin/cluster/decommission/awareness/get/GetDecommissionStateRequest.java:55]: nit: Strings.isEmpty

**Review by Bukhtawar — APPROVED** [review]

**Pull request merged**